File: fixtures/typescript/typescript-modules/packages/app/src/greeting-service.ts
import { GreetingPort } from '@minos/greeting-api'

export class GreetingService {
  constructor(private readonly port: GreetingPort) {}

  greet(name: string): string
  greet(names: readonly string[]): string[]
  greet(input: string | readonly string[]): string | string[] {
    if (typeof input === 'string') {
      return this.port.greet(input)
    }
    return input.map(name => this.port.greet(name))
  }
}

File: fixtures/java/java-multi-module/app/src/main/java/io/github/fturleque/minos/fixture/multimodule/app/GreetingService.java
package io.github.fturleque.minos.fixture.multimodule.app;

import io.github.fturleque.minos.fixture.multimodule.api.GreetingPort;
import io.github.fturleque.minos.fixture.multimodule.api.UserProfile;

import java.util.Objects;

public final class GreetingService {

    private final GreetingPort greetingPort;

    public GreetingService(GreetingPort greetingPort) {
        this.greetingPort = Objects.requireNonNull(greetingPort, "greetingPort");
    }

    public String greet(String name) {
        return greetingPort.greet(new UserProfile(name));
    }
}
